### ISSUE TITLE
Flowchart Event Handlers trigger when FlowChart MonoBehaviour is disabled.

### ISSUE THREAD

**Issue by awiebe** [snozbot/fungus#679]
Flowcharts do not trigger if the game-object they are attached to is inactive, but still trigger if the component is not enabled.

My use case is I want to enable the flowchart with a keypress start condition after the player enters a proximity trigger area.  A operate script would be used to enable the flowchart component, but as it stands it seems  I have to put the flowchart on a separate game object to the intractable(which seems inconsistent with say switching to a trigger event handler which expects to be on the trigger object).

Wherever the `Flowchart.cs` start processing blocks needs a check for `MonoBehaviour.enabled`

**Comment by stevehalliwell** [comment]
Makes sense, as every block and event and command is its own monobeh, just hidden, disabling the flowchart itself isn't stopping them from acting. 

I'm happy to do up a PR to prevent EventHandlers from firing if the flowchart is disabled. 
Unless you're already working on one?

**Comment by awiebe** [comment]
I tried to patch it myself but it seems to have still fired anyway.  I tried using a keypress(space) event handler as trigger, and it seemed to fire the flowchart even though I tried to check for enabled in the flowchart component before trying to process block.

I guess I must not fully understand the path from the event handler to the flowchart, so feel free to work on it, because I'm stumped.
